app/assets/javascripts/quotingPanel.js

```diff
@@ -85,13 +114,58 @@ class QuotingPanel {
   addListeners() {
     document.getElementById("SubmitQuote").addEventListener('submit', (event) => { event.preventDefault() });
     document.getElementById("searchAddressButton").addEventListener('click', this.handleSearchAddressClick);
+    document.getElementsByName("serviceExpeditionCost").forEach((element) => { 
+      element.addEventListener('click', (event) => { 
+        const serviceExpeditionCost = Number.parseFloat(event.target.value);
+        const serviceExpeditionDetails = event.target.dataset.label;
+        const serviceExpeditionTime = event.target.id;
+        this.handleServiceExpeditionChange(serviceExpeditionCost, serviceExpeditionDetails, serviceExpeditionTime); 
+      });
+    });
+    document.getElementById("addBag").addEventListener('click', (event) => { this.handleAddSaltBag() });
+    document.getElementById("removeBag").addEventListener('click', (event) => { this.handleRemoveSaltBag() });
+  }
+
+  handleServiceExpeditionChange(serviceExpeditionCost, serviceExpeditionDetails, serviceExpeditionTime) {
+    this.serviceExpeditionCost = serviceExpeditionCost;
+    this.serviceExpeditionDetails = serviceExpeditionDetails;
+    this.serviceExpeditionTime = serviceExpeditionTime;
+    this.totalDue = this.calculateTotalDue();
```

Comment:
Shouldn't this be done as part of `updateTotalDueNode`?

Reply:
refactored in a function that runs both calculate and update node